src/town_cmd.cpp

```diff
@@ -2036,6 +2039,9 @@ static void DoCreateTown(Town *t, TileIndex tile, uint32_t townnameparts, TownSi
 		GrowTown(t);
 	} while (--i);
 
+	t->UpdateVirtCoord();
```

Comment:
Moving this does not change the state of `_generating_town` when these lines are called, unless I'm reading something wrong.

Reply:
These are moved so that it's done after the town growth is completed instead of before. `_generating_town` is tested only in `ChangePopulation()`, not these two calls.

Reply:
Oh, I see -- this does the update that is skipped during `ChangePopulation()`.